app/scripts/controllers/onboarding.ts

```diff
@@ -0,0 +1,108 @@
+import { ObservableStore } from '@metamask/obs-store';
+import log from 'loglevel';
+import { FirstTimeFlowType } from '../../../shared/constants/onboarding';
+
+/**
+ * The state of the OnboardingController
+ */
+export type OnboardingControllerState = {
+  seedPhraseBackedUp: boolean | null;
+  firstTimeFlowType: FirstTimeFlowType | null;
+  completedOnboarding: boolean;
+  onboardingTabs?: Record<string, string>;
+};
+
+const defaultTransientState = {
+  onboardingTabs: {},
+} satisfies Pick<OnboardingControllerState, 'onboardingTabs'>;
+
+const defaultState = {
+  seedPhraseBackedUp: null,
+  firstTimeFlowType: null,
+  completedOnboarding: false,
+} satisfies OnboardingControllerState;
+
+/**
+ * Controller responsible for maintaining
+ * state related to onboarding
+ */
+export default class OnboardingController {
+  /**
+   * Observable store containing controller data.
+   */
+  store: ObservableStore<OnboardingControllerState>;
+
+  /**
+   * Constructs a Onboarding  controller.
+   *
+   * @param options - the controller options
+   * @param options.initState - Initial controller state.
+   */
+  constructor({ initState }: { initState: OnboardingControllerState }) {
```

Comment:
It looks like this would be a breaking change, since the compiler would raise an error if `initState` doesn't contain all properties of `OnboardingControllerState`.

How about adding a `Partial` type, so that it's possible to only update some properties of `defaultState`?

```suggestion
  constructor({ initState }: { initState: Partial<OnboardingControllerState> }) {
```